### PR TITLE
Statistics page implementation

### PR DESCRIPTION
**Pull request by fordelord**
/statistics page
- Add statistics page
- Implement filter by years
- Add minimal mobile styles

### CODE REVIEW

File: components/Statistic/MonthlyDiagram.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const { monthlyDiagramWrapper, monthlyDiagramColumn } = styles(theme);
+
+  return (
+    <Box sx={monthlyDiagramWrapper}>
+      {seasonData.map((column) => (
+        <Box sx={monthlyDiagramColumn} key={column.month}>
+          <Typography
+            sx={{
+              ...counterMarginBottom(postsByMonth[column.month]),
+            }}
```
Comment (1alexvash) [inline components/Statistic/MonthlyDiagram.tsx:39]:
```suggestion
            sx={
              counterMarginBottom(postsByMonth[column.month]),
            }
```

File: components/Statistic/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    backgroundColor: condition
+      ? theme.palette.main.orange
+      : theme.palette.mode === "light"
+      ? theme.palette.main.lightGrey
+      : theme.palette.main.grey,
+    color: condition
+      ? theme.palette.main.white
+      : theme.palette.mode === "light"
+      ? theme.palette.main.black
+      : theme.palette.main.white,
```
Comment (1alexvash) [inline components/Statistic/index.tsx:43]: nested ternaries

File: components/Statistic/pageHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+export const getPostsByYearAndMonth = (
+  year: number,
+  posts: PostDocumentWithoutBody[]
+) => {
+  const postsByMonth: PostsByMonthType = {};
+  const filteredPosts = posts.filter(
+    (post) => new Date(post.date).getFullYear() === year
+  );
+
+  filteredPosts.map((post) => {
```
Comment (1alexvash) [inline components/Statistic/pageHelpers.ts:26]: https://alex-code.com/post/2023-05-07-whats-the-difference-beetween-map-and-forEach-methods

File: components/Statistic/pageHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  const sortedMonths = Object.keys(postsByMonth).sort((a, b) => {
+    const monthA = new Date(`2000 ${a}`).getMonth();
+    const monthB = new Date(`2000 ${b}`).getMonth();
+    return monthA - monthB;
+  });
+
+  const sortedPostsByMonth: { [month: string]: number } = {};
+  sortedMonths.map((month) => {
+    sortedPostsByMonth[month] = postsByMonth[month];
+  });
```
Comment (1alexvash) [inline components/Statistic/pageHelpers.ts:41]: https://alex-code.com/post/2023-05-07-whats-the-difference-beetween-map-and-forEach-methods

File: pages/statistics.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+const Statistics = ({ posts }: Props) => (
+  <>
+    <Head>
+      <title>{config.site_title}</title>
+      <meta property="og:title" content={config.site_keywords[1]} />
+      <meta property="og:description" content={config.site_description} />
+      <meta property="og:url" content={config.host_url} />
+      <meta property="og:type" content="website" />
+      <meta property="og:site_name" content={config.site_title} />
+      <meta name="description" content={config.site_description} />{" "}
```
Comment (1alexvash) [inline pages/statistics.tsx:43]:
```suggestion
      <meta name="description" content={config.site_description} />
```

File: pages/statistics.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+const Statistics = ({ posts }: Props) => (
+  <>
+    <Head>
+      <title>{config.site_title}</title>
+      <meta property="og:title" content={config.site_keywords[1]} />
+      <meta property="og:description" content={config.site_description} />
+      <meta property="og:url" content={config.host_url} />
+      <meta property="og:type" content="website" />
+      <meta property="og:site_name" content={config.site_title} />
```
Comment (1alexvash) [inline pages/statistics.tsx:42]: Maybe this page does not need meta

File: pages/statistics.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+import client from ".tina/__generated__/client";
+
+interface Props {
+  posts: PostDocumentWithoutBody[];
+}
+
+const containterStyles = {
+  mr: "auto",
+  ml: "auto",
```
Comment (1alexvash) [inline pages/statistics.tsx:20]: mx

File: pages/statistics.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  pl: "15px",
+  pr: "15px",
```
Comment (1alexvash) [inline pages/statistics.tsx:22]: px

File: components/Statistic/pageHelpers.ts (hunk 4 above)
Comment (1alexvash) [reply components/Statistic/pageHelpers.ts:46]: @fordelord you pressed a thumbs up, and probbably read the article, but didn't alter your code

So here goes my question:

What are you return on the line **44**?

File: components/Statistic/pageStyles.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+      alignItems: "end",
+      width: "100%",
+      background: theme.palette.main.orange,
+      "& img": {
+        mb: "3px",
+        "@media (max-width: 1019px)": {
+          width: "100%",
+          height: "100%",
+          maxHeight: "30px",
+          minHeight: "30px",
```
Comment (1alexvash) [inline components/Statistic/pageStyles.ts:66]: Should you specify both?

File: components/Statistic/pageStyles.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    "& .MuiPaper-root": {
+      display: "flex",
+      justifyContent: "center",
+      alignItems: "end",
+      width: "100%",
+      background: theme.palette.main.orange,
+      "& img": {
+        mb: "3px",
+        "@media (max-width: 1019px)": {
```
Comment (1alexvash) [inline components/Statistic/pageStyles.ts:62]: what this media query means?

has it been used in other places?

File: components/Statistic/pageStyles.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    "& .MuiBox-root": {
+      color:
+        theme.palette.mode === "light"
+          ? theme.palette.main.black
+          : theme.palette.main.white,
+      fontSize: "14px",
+      width: "100%",
+      maxWidth: "90px",
+      textAlign: "center",
+      "@media (max-width: 1200px)": {
```
Comment (1alexvash) [inline components/Statistic/pageStyles.ts:87]: Do you need media queries for this page?

Is it possible to use default CSS fluid styles?

File: pages/statistics.tsx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+import client from ".tina/__generated__/client";
+
+interface Props {
+  posts: PostDocumentWithoutBody[];
+}
+
+const containterStyles = {
+  mx: "auto",
+  px: "15px",
+  ["@media(min-width: 768px)"]: {
```
Comment (1alexvash) [inline pages/statistics.tsx:21]:
```suggestion
  ["@media (min-width: 768px)"]: {
```
watch out for small details like this

File: pages/statistics.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+    width: "750px",
+  },
+  ["@media(min-width: 1020px)"]: {
+    width: "1000px",
+  },
+  ["@media(min-width: 1260px)"]: {
+    width: "1230px",
+  },
```
Comment (1alexvash) [inline pages/statistics.tsx:29]: Do you need so many breakpoints in here?

Can the page just be fluid / flexible?

File: redux/slices/theme.ts [hunk 14]
```diff
@@ -2,7 +2,7 @@ import { createSlice, PayloadAction } from "@reduxjs/toolkit";
 
 export type Theme = "light" | "dark";
 
-export const defaultTheme = "dark";
+export const defaultTheme = "light";
```
Comment (1alexvash) [inline redux/slices/theme.ts:5]: ?

**Review by 1alexvash — CHANGES_REQUESTED** [review]
UI looks
Code quality is very questionable

File: components/Statistic/MonthlyDiagramStatistic.tsx [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+                src={`/images/${column.season}.svg`}
+                alt={`${column.season}`}
+                height={50}
+                width={50}
+              />
+            </Paper>
+          </Box>
+        ))}
+      </Box>
+      <Box sx={monthTitlesStyle}>
```
[the remark this answers is not in the document]
Comment (fordelord) [reply components/Statistic/MonthlyDiagramStatistic.tsx:64]: When I used the name you said bad name, title is bad too) Ok return to `monthName` without style suffix

File: components/Statistic/index.tsx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+      color,
+    };
+  };
+
+  const pageControls = (
+    <Box
+      sx={[
+        pageControlsAlign,
+        {
+          "& div": yearButton,
```
[the remark this answers is not in the document]
Comment (fordelord) [reply components/Statistic/index.tsx:71]: Change `yearButton` to `yearButtonStyle`

File: redux/slices/theme.ts (hunk 14 above)
Comment (fordelord) [reply redux/slices/theme.ts:5]: We have the different default themes in MUI and in style of our project. You add the `.dark-theme` class after the check of localStorage. That it why you have the blinking in light theme in components from MUI

File: components/Statistic/pageHelpers.ts [hunk 17]
```diff
@@ -0,0 +1,49 @@
+import { PostDocumentWithoutBody } from "interfaces";
+
+import { PostsByMonthType } from ".";
+
+export const marginBottomCounter = (postsCount: number | undefined) => {
```
[the remark this answers is not in the document]
Comment (fordelord) [reply components/Statistic/pageHelpers.ts:5]: Fixed

**Review by 1alexvash — APPROVED** [review]

**Pull request merged**